### ISSUE TITLE
Data overwriting

### ISSUE THREAD

**Issue by pipliggins** [globaldothealth/isaric#203]
In instances where data is written over multiple lines for each subject (e.g. a row for admission, a row for outcome, and a row for each day they were in hospital), data is being overwritten in certain instances.

E.g. for antiviral type with schema code

```
[visit.treatment_antiviral_type]
    combinedType = "set"
    excludeWhen = "none"
    fields = [
      { field = "daily_antiviral_type___1", values = { 1 = "Ribavirin" } },
      { field = "daily_antiviral_type___2", values = { 1 = "Lopinavir/Ritonavir" } },
      { field = "daily_antiviral_type___3", values = { 1 = "Interferon alpha" } },
      { field = "daily_antiviral_type___4", values = { 1 = "Interferon beta" } },
      { field = "overall_antiviral_dc___1", values = { 1 = "Ribavirin" } },
      { field = "overall_antiviral_dc___2", values = { 1 = "Lopinavir/Ritonavir" } },
      { field = "overall_antiviral_dc___3", values = { 1 = "Interferon alpha" } },
      { field = "overall_antiviral_dc___4", values = { 1 = "Interferon beta" } },
...
      ]
```
if the 'overall' columns are filled only once, and in an earlier row than the daily columns for a given patient, adtl returns whichever antiviral was administered last in the data table, not a combined list which accumulates across rows. This is an issue previously touched on in #131 (got around by adding the dates to each row in the data) but is more of an issue here.

2 Options:
   1. Only use the 'overall' variables in this case. These are only filled in once, so no overwriting risk, and should contain all the data. The risk is that if these were completed manually by clinicians, administered drugs might have been missed which would be caught by mapping the daily values.
   2. Edit adtl to store the long format tables for each group ID and merge them during data parsing, rather than assuming 1 row per patient which allows overwriting. 

May also need thinking about in the context of moving to a new format... when parsing data into FHIR/OMAP this issue may disappear.

I noticed this when looking at the Malaysia dataset, but I'm sure it's an issue in other parsers as well.

**Comment by pipliggins** [comment]
@abhidg @sadiekelly thoughts on approaches to this?

**Comment by sadiekelly** [comment]
hi @pipliggins! so if daily_type___1 is indicated (Ribavirin) and overall___4 also indicated (Interferon beta) only Interferon beta would show in the adtl? Also if both overall___2 and overall___4 are indicated only overall___4 would show? Or is it only specific for the daily variables that are on different rows in the raw data?

Regarding the daily vs overall completion, I had a look at the Malaysia data and the antiviral type variables completed at daily and overall, to see to what extent daily variables are indicated as 'medication received' (response = 1) when there is either no indication (form not completed) at overall, or the response at overall is 'medication not received' (0) (although to also remember these are checkbox variables so '0' doesn't necessarily mean 'not received'). There are just under 200 cases where a medication is indicated as received in a daily form but indicated as not received in the overall form. Additionally there are 50 cases where medication is indicated as received in a daily form but the overall form has not been completed. So we could miss these ~250 cases by not mapping the daily variables, representing 1.5 % of the total number of records with data available for these variables. It is a more common occurrence to have '0' for the daily variable and '1' in the overall form (7244 cases). This refers to a dataset of ~6500 patients so the number of cases missed could scale up if this occurs frequently across many of the datasets, which I think is very likely. 

My thoughts are that we should try to capture as much as we can, so that if a medication is indicated in a daily form but not at overall (or vice versa) then this should be recorded so that all instances of medication received are represented. There could be many reasons why either the daily or the overall form could be incorrect or not completed across the different datasets and I don't think we should apply a blanket decision for all datasets and make a decision on determining the source of truth during curation of the data.

**Comment by pipliggins** [comment]
| Subjid | redcap_day | daily_type___1 | daily_type___2 | daily_type___3 | overall_type___1 | overall_type___2 | overall_type___3 |
|--------|------------|----------------|----------------|----------------|------------------|------------------|------------------|
| 1      | admission  |                |                |                |                  |                  |                  |
| 1      | discharge  |                |                |                | 1                | 0                | 1                |
| 1      | day1       | 1              | 0              | 0              |                  |                  |                  |
| 1      | day2       | 0              | 0              | 1              |                  |                  |                  |
| 2      | admission  |                |                |                |                  |                  |                  |

For this simplified data, for subject 1 adtl would only output "Interferon alpha" in the final civ, rather than the correct ["Ribavarin", "Interferon alpha"] output, because it's only saving the final row (or, more accurately, the last *non null* data for each field). 

I would tend to agree about trying to save the data...

**Comment by abhidg** [comment]
This is an issue with adtl, will have a think about what's the best way to resolve this.

For FHIR, this is less of an issue as each row represents one fact, so Interferon alpha and Ribavirin would be on different rows, something that adtl should be able to do with a oneToMany type. This would raise the issue of duplicates, if the medication is mentioned in both daily and overall, but that is an easier fix.

**Comment by pipliggins** [comment]
Closed by globaldothealth/adtl#90

**Comment by pipliggins** [comment]
Closed by globaldothealth/adtl#90
